papermage/magelib/document.py

```diff
@@ -44,43 +46,49 @@ def __init__(self, symbols: str, metadata: Optional[Metadata] = None):
     def fields(self) -> List[str]:
         return list(self.__entity_span_indexers.keys()) + self.SPECIAL_FIELDS
 
-    def find_by_span(self, query: Union[Entity, Span], field_name: str) -> List[Entity]:
-        if isinstance(query, Entity):
-            return self.__entity_span_indexers[field_name].find(query=query)
-        elif isinstance(query, Span):
+    def find(self, query: Union[Span, Box], field_name: str) -> List[Entity]:
+        if isinstance(query, Span):
             return self.__entity_span_indexers[field_name].find(query=Entity(spans=[query]))
-        else:
-            raise TypeError(f"Unsupported query type {type(query)}")
-
-    def find_by_box(self, query: Union[Entity, Box], field_name: str) -> List[Entity]:
-        if isinstance(query, Entity):
-            return self.__entity_box_indexers[field_name].find(query=query)
         elif isinstance(query, Box):
             return self.__entity_box_indexers[field_name].find(query=Entity(boxes=[query]))
         else:
             raise TypeError(f"Unsupported query type {type(query)}")
 
+    def find_by_span(self, query: Entity, field_name: str) -> List[Entity]:
+        # TODO: will rename this to `intersect_by_span`
+        return self.__entity_span_indexers[field_name].find(query=query)
+
+    def find_by_box(self, query: Entity, field_name: str) -> List[Entity]:
+        # TODO: will rename this to `intersect_by_span`
+        return self.__entity_box_indexers[field_name].find(query=query)
+
     def check_field_name_availability(self, field_name: str) -> None:
         if field_name in self.SPECIAL_FIELDS:
             raise AssertionError(f"{field_name} not allowed Document.SPECIAL_FIELDS.")
         if field_name in self.__entity_span_indexers.keys():
-            raise AssertionError(f"{field_name} already exists. Try `is_overwrite=True`")
+            raise AssertionError(f"{field_name} already exists. Try `doc.remove_entity({field_name})` first.")
```

Comment:
```suggestion
            raise AssertionError(f"{field_name} already exists. Try `doc.remove_entity(\"{field_name}\")` first.")
```

Nit, but otherwise it looks like you are providing a variable.

Reply:
done